src/main/kotlin/inc/evil/d17map/config/CachedBodyHttpServletRequest.kt

```diff
@@ -0,0 +1,26 @@
+package inc.evil.d17map.config
+
+import jakarta.servlet.ServletInputStream
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletRequestWrapper
+import org.springframework.util.StreamUtils
+import java.io.*
+
+class CachedBodyHttpServletRequest(request: HttpServletRequest) : HttpServletRequestWrapper(request) {
+
+    private val cachedBody: ByteArray
+
+    init {
+        val requestInputStream = request.inputStream
+        cachedBody = StreamUtils.copyToByteArray(requestInputStream)
```

Comment:
Może warto byłoby w takich miejscach pododawać jakieś logi lub podstawowy error handling, żeby w razie wystąpienia na tym etapie jakichś błędów można było to łatwo wyłapać

````
try {
        val requestInputStream = request.inputStream
        cachedBody = StreamUtils.copyToByteArray(requestInputStream)
    } catch (e: IOException) {
        throw RuntimeException("Failed to cache request body", e)
    }

Reply:
Czyli po prostu tutaj chciałabyś zmienić message wyjątku który poleci, żebyśmy wiedzieli, że to stąd?
Zastanawiam się, czy loggowanie by nie wystarczyło w takim wypadku?

Bo case jes taki, że jeśli tutaj rzuci się wyjątek, to on nie trafi do tego `@ControllerAdvice` w naszym `GlobalHandlerze`. Ten wyjątek sobie poleci w odmęty Springa, gdzieś do obsługi Servletów Javowych - filtr który  korzysta z tej klasy jest jednym z pierwszych filtrów jakie dostaną request, czyli to jest coś typu `fun main` -> `coś z Servletów` -> `filterchain` -> `filtr korzystający z tej klasy`. Więc do naszej logiki biznesowej ten request nawet nie dojdzie jeśli wyjątek poleci. Żadne Security się nie wykona ani nic.

Stąd pytanie, co nam daje tutaj rzucenie innego wyjątku? Wydaje mi się, że dodanie loggera by wystarczyło.

Reply:
Faktycznie - bardziej niż to miałam na celu dodanie jakiegoś bardziej opisującego message, żebyśmy mieli lepszą świadomość co się wydarzyło

Przekombinowałam, zignoruj to 🫢

Reply:
To dodam logi

Reply:
Done